module/VuFind/src/VuFind/Db/Entity/SearchEntityInterface.php

```diff
@@ -40,4 +42,170 @@
  */
 interface SearchEntityInterface extends EntityInterface
 {
+    /**
+     * Get identifier (returns null for an uninitialized or non-persisted object).
+     *
+     * @return ?int
+     */
+    public function getId(): ?int;
+
+    /**
+     * Get user.
+     *
+     * @return int
+     */
+    public function getUserId(): int;
+
+    /**
+     * Set user.
+     *
+     * @param int $user User
+     *
+     * @return SearchEntityInterface
+     */
+    public function setUserId(int $user): SearchEntityInterface;
+
+    /**
+     * Get session.
+     *
+     * @return ?string
+     */
+    public function getSessionId(): ?string;
+
+    /**
+     * Set session.
+     *
+     * @param ?string $session_id Session id
+     *
+     * @return SearchEntityInterface
+     */
+    public function setSessionId(?string $session_id): SearchEntityInterface;
+
+    /**
+     * Get created.
+     *
+     * @return DateTime
+     */
+    public function getCreated(): DateTime;
+
+    /**
+     * Set created date.
+     *
+     * @param DateTime $dateTime Created date
+     *
+     * @return SearchEntityInterface
+     */
+    public function setCreated(DateTime $dateTime): SearchEntityInterface;
+
+    /**
+     * Get title.
+     *
+     * @return ?string
+     */
+    public function getTitle(): ?string;
+
+    /**
+     * Set title.
+     *
+     * @param ?string $title Title
+     *
+     * @return SearchEntityInterface
+     */
+    public function setTitle(?string $title): SearchEntityInterface;
+
+    /**
+     * Get saved.
+     *
+     * @return int
+     */
+    public function getSaved(): int;
```

Comment:
While the "saved" value is an integer in the database definition, we use it as a boolean (since MySQL doesn't support native booleans). It would probably make sense to change these methods to take and return boolean values. In the actual getter and setter you can convert -- i.e. `$this->saved = $saved ? 1 : 0;` and `return (bool)($this->saved ?? 0);`.